assets/js/components/Dashboard/AlertsPage.tsx

```diff
@@ -267,6 +278,22 @@ const AlertsList: ComponentType<AlertsListProps> = ({
     return Date.parse(end1) - Date.parse(end2);
   };
 
+  const getNumberOfAffectedPlaces = (screensByAlert: string[]) => {
+    // Using a Set so we don't have to worry about counting a place more than once.
+    const placesCounted = new Set();
+
+    places.forEach((place) => {
+      if (
+        place.screens.filter((screen) => screensByAlert.includes(screen.id))
+          .length > 0
+      ) {
+        placesCounted.add(place.id);
+      }
+    });
+
+    return placesCounted.size;
+  };
```

Comment:
`getNumberOfAffectedPlaces` and `placesWithAffectedAlert` seem to have fairly similar logic. I know they're a little different (one returns a list of the actual places and is meant for a single selected alert while the other just returns the count of the affected places), but I'm wondering if there's a way to consolidate the logic for the sake of maintainability.

If it seems like more trouble than it's worth though, it's probably okay to leave it as is since I can't really foresee a business reason for either logic to change in the future.

Reply:
Good catch. I was avoiding that function at first because it looked at component state, but it's super easy to change it so it takes state as a variable instead.

Reply:
Ah nice. Glad it was an easy tweak!